### PR TITLE
Compile same structure for app.json as compiled by ZAM

### PR DESCRIPTION
**Pull request by Mehul-Porwal**
## Description

ZAM and ZAF have introduced some changes to the json structure of a compile app. Making changes here to comply with the new structure.

@zendesk/vegemite 

## Checklist

- [ ] :guardsman: includes new unit and functional tests

### CODE REVIEW

**Review by token-cjg — APPROVED** [review]

File: packages/zcli-apps/src/types.ts [hunk 1]
```diff
@@ -97,7 +96,7 @@ export interface Installation {
     id: string;
     plan?: string;
     requirements: {};
-    settings: Settings;
+    settings: Array<any>;
```
Comment (etanxing) [inline packages/zcli-apps/src/types.ts:99]: Can be `Array<Record<string, any>>` like other places and the old interface `Settings` can be removed.

**Review by cl8au — APPROVED** [review]

File: packages/zcli-apps/src/lib/buildAppJSON.ts [hunk 2]
```diff
@@ -84,6 +107,16 @@ export const getAppPayloadFromManifest = (appManifest: Manifest, port: number, a
   }
 }
 
+const getSettingsArr = (appSettings: any) => {
+  const s: any[] = []
+  Object.keys(appSettings).forEach((settingName: string) => {
+    const setting: Record<string, any> = {}
```
Comment (etanxing) [inline packages/zcli-apps/src/lib/buildAppJSON.ts:113]: I saw `settings` are coming from two places so I think it's safe to use `const setting: Record<string, string> = {}`
```
  const configSettings = manifest.parameters.reduce((result: Dictionary<string>, param) => {
    if (configContainsParam(param.name)) {
      result[param.name] = configParams[param.name] as string
    }
    return result
  }, {})
```

```
  const settings: Dictionary<string> = {}
  for (const param of params) {
    const value = await cli.prompt(`Enter ${appName} setting.${param.name} value`, { type: param.secure ? 'hide' : 'normal', required: valuesRequired })
    if (value) settings[param.name] = value
  }
  return settings
```

File: packages/zcli-apps/src/lib/buildAppJSON.ts [hunk 3]
```diff
@@ -84,6 +107,16 @@ export const getAppPayloadFromManifest = (appManifest: Manifest, port: number, a
...
+    setting[settingName.toString()] = appSettings[settingName.toString()]
```
Comment (etanxing) [inline packages/zcli-apps/src/lib/buildAppJSON.ts:114]: We see `settingName` as `string` type so I think we don't need `toString()` here.

Comment (Mehul-Porwal) [reply packages/zcli-apps/src/lib/buildAppJSON.ts:114]: typescript was complaining and this had to be enforced here.

**Review by etanxing — APPROVED** [review]

**Pull request merged**